### PR TITLE
Report errors parsing geoshape, geotrace, and points values

### LINKED ISSUE #309: Geo classes ignore bad inputs
#### Software versions 
JavaRosa v2.9.0

#### Problem description
GeoPointData (and probably GeoShapeData and GeoTraceData) are very “accepting”—to be generous—in the inputs they process. All sorts of garbage is silently turned into 0s.

#### Steps to reproduce the problem
Run GeoDistanceTest. 
Change the numbers in this section of distance.xml to invalid values (such as non-numeric ones) and run again.
```
<geoshape>0 0; 0 1; 0 91</geoshape>
<geotrace>0 0; 0 1; 0 91</geotrace>
<points>
    <point>0 0</point>
    <point>0 1</point>
    <point>0 91</point>
</points>
```

#### Expected behavior
There should be error messages about invalid inputs.

### PR DESCRIPTION
**Pull request by dcbriccetti**
Closes #309

#### What has been done to verify that this works as intended?
I observed that the new message appears in the log. I tested the change in Collect with forms containing all three geo data types, to ensure there is no harmful impact.

#### Why is this the best possible solution? Were any other approaches considered?
This logs a message with the offending data.

#### Are there any risks to merging this code? If so, what are they?
The only impact is on what appears in the log.

Example error messages from a test run:
```
INFO org.javarosa.xform.parse.XFormParser - Title: "Invalid geoshape"
WARN XFormAnswerDataParser - Could not create GeoShapeData from "one; 2"
WARN XPathFuncExprGeo - The function 'distance' received a value that does not represent GPS coordinates
```

### CODE REVIEW

File: resources/org/javarosa/core/util/distance.xml [hunk 1]
```diff
@@ -7,11 +7,13 @@
...
-                    <geoshape>0 0; 0 91</geoshape>
-                    <geotrace>0 0; 0 91</geotrace>
+                    <!-- 0 is used internally for “no data”, so we’ll use 1e-15, a slightly larger number that still
+                         works for our calculations around the equator -->
+                    <geoshape>1e-15 1; 1e-15 91</geoshape>
+                    <geotrace>1e-15 1; 1e-15 91</geotrace>
                     <points>
-                        <point>0 0</point>
-                        <point>0 91</point>
+                        <point>1e-15 1</point>
```
[the remark this answers is not in the document]
Comment (dcbriccetti) [reply resources/org/javarosa/core/util/distance.xml]: It appears someone chose that value arbitrarily. It *is* extremely unlikely that real data would be exactly 0, but still ....

File: src/org/javarosa/xform/util/XFormAnswerDataParser.java [hunk 2]
```diff
@@ -46,150 +46,131 @@
...
+                return d == null ? null : new DateData(d);
+
+            case TIME:
+                Date t = trimmedText == null ? null : DateUtils.parseTime(trimmedText);
+                return t == null ? null : new TimeData(t);
+
+            case BOOLEAN:
+                if (trimmedText == null) {
+                    return null;
+                } else {
```
[the remark this answers is not in the document]
Comment (dcbriccetti) [reply src/org/javarosa/xform/util/XFormAnswerDataParser.java]: I agree, but mine were just formatting changes, so I think I’ll stay out of that.

File: src/org/javarosa/xform/util/XFormAnswerDataParser.java [hunk 3]
```diff
@@ -46,150 +46,131 @@
...
+                selections.add(s);
         }
 
-        return v;
+        return selections;
     }
 
     private static Selection getSelection(String choiceValue, QuestionDef q) {
-        Selection s;
+        final Selection s;
```
[the remark this answers is not in the document]
Comment (dcbriccetti) [reply src/org/javarosa/xform/util/XFormAnswerDataParser.java]: I agree! Also here, I made no changes that would require testing, so we can save this one for later.

File: resources/org/javarosa/core/util/distance.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggalmazor) [reply resources/org/javarosa/core/util/distance.xml]: Would it be crazy to switch places? Using 1e-15 for missing data?

File: src/org/javarosa/xform/util/XFormAnswerDataParser.java [hunk 4]
```diff
@@ -128,6 +132,7 @@ public static IAnswerData getAnswerData(String text, int intDataType, QuestionDe
                 try {
                     return new GeoPointData().cast(new UncastData(trimmedText));
                 } catch (Exception e) {
+                    logGeoCreateError(GeoPointData.class, trimmedText);
```
[the remark this answers is not in the document]
Comment (dcbriccetti) [reply src/org/javarosa/xform/util/XFormAnswerDataParser.java]: I agree. I’ll add that to the logging.

**Comment by dcbriccetti** [comment]
Thanks for reviewing, @ggalmazor.

**Pull request merged**